Abstraction contrasts on utils_contrasts

ferponcem opened the pull request:
Now for real, sorry for the messy steps!
This PR contains some changes made to `utils_contrasts` in order to add the contrasts we discussed [here](https://github.com/hbp-brain-charting/public_analysis_code/blob/feature/contrasts_descriptions/ibc_data/contrasts/Abstraction.csv) for **Abstraction** task. 
@bthirion could you have a look?

bthirion commented on ibc_public/utils_contrasts.py:
```diff
@@ -278,10 +280,143 @@ def multi_modal(design_matrix_columns):
 
+def abstraction(design_matrix_columns):
+    """ Contrasts for Abstraction """
+    localizer_ = ['localizer_faces', 'localizer_humanbody',
```
Localizer should be another block I think

bthirion commented on ibc_public/utils_contrasts.py:
```diff
@@ -278,10 +280,143 @@ def multi_modal(design_matrix_columns):
+               'localizer_numbers-others',
+               'localizer_places-others', 'localizer_objects-others',
+               'localizer_checkerboards-others','humanbody-other',
+               'animals-other','faces-other',
```
be consistent on naming: 'other' vs 'others'

bthirion commented on ibc_public/utils_contrasts.py:
```diff
@@ -278,10 +280,143 @@ def multi_modal(design_matrix_columns):
+    contrasts['flora-other'] = 5 * flora - allstim
+    contrasts['places-other'] = 5 * places - allstim
+    contrasts['objects-other'] = 5 * objects - allstim
+    contrasts['geometry-others'] = 2 * geometry - allstim
```
```suggestion
    contrasts['geometry-others'] = 3 * geometry - allstim
``` as we want the contrasts to sum to 0 ?

bthirion replied:
Same thing for the following contrasts.

ferponcem commented:
Thanks for the feedback!
Some changes I made:
* Separated the localizer as a different block, now we have both `Abstraction` and `AbstractionLocalizer`. See changes in the script. Along to that I made the changes on the [conditions file](https://github.com/hbp-brain-charting/public_analysis_code/blob/master/ibc_data/ibc_conditions.tsv#L389), on the contrast [sheet](https://github.com/hbp-brain-charting/public_analysis_code/blob/feature/contrasts_descriptions/ibc_data/contrasts/Abstraction.csv) and added a [new one](https://github.com/hbp-brain-charting/public_analysis_code/blob/feature/contrasts_descriptions/ibc_data/contrasts/AbstractionLocalizer.csv) for the localizer.
* Modified the weights for the contrasts as we want them to sum up to zero,hopefully like this is alright for this format!
* For the sake of consistency, I am using `other` for the contrasts names

Let me know what you think @bthirion !

bthirion commented on ibc_public/utils_contrasts.py:
```diff
@@ -279,9 +283,158 @@ def multi_modal(design_matrix_columns):
+        con['localizer_humanbody'] - localizer
+    contrasts['localizer_words-other'] = 8 * con['localizer_words'] -\
+        localizer
+    contrasts['localizer_nonsensewords-other'] = 7 *\
```
Why 7 ?

ferponcem replied:
Fixed! Thanks!

bthirion reviewed (commented):
LGTM otherwise.

bthirion reviewed (approved):
LGTM, thx.
